### PR TITLE
`[Whitelisting]` Root branch

### LINKED ISSUE #2470: Implement Whitelisting UI / UX

### PR DESCRIPTION
**Pull request by mudrila**
# Changes

- Refactor components structure
  - Remove `src/components/pages` - extract underlying components right into `src/components`
  - Better replicate routes structure within `src/pages` - move all the components that are relates to Safe's-related pages into `src/pages/dao`
  - Rename all the components under `src/pages/dao` from `DAO<Stuff>` to `Safe<Stuff>`
  - Update routes accordingly
- Implement multiple strategies support
  - Properly fetch and map proposals when there're multiple strategies attached
  - Properly vote on proposal when there're multiple strategies attached
  - Properly verify whether user is able to create proposal, assuming there might be multiple strategies attached
- Implement `Roles Whitelisting`
  - Whitelisting / un-whitelisting for the role
  - Creating proposal from the whitelisted role's member
  - Displaying permissions badge on the role
- Updating settings page
  - New Safe Settings layout
  - Introducing global "Proposal Actions Builder" page pattern
  - Implementing Permissions on the Safe with Azorius module attached
    - UI for updating permissions when there's ERC20 strategy attached / not attached
    - UI for updating permissions when there's ERC721 strategy attached / not attached
    - UI / UX / functionality for updating / creating permissions when there's ERC20 With Whitelisting strategy attached
    - UI / UX / functionality for updating / creating permissions when there's ERC721 With Whitelisting strategy attached

## Issues
- Closes #2470

### CODE REVIEW

File: src/models/FreezeGuardTxBuilder.ts [hunk 1]
```diff
@@ -213,16 +213,16 @@ export class FreezeGuardTxBuilder extends BaseTxBuilder {
   private setFreezeVotingAddress() {
     let freezeVotingByteCodeLinear: Hex;
     if (this.parentStrategyType) {
-      if (this.parentStrategyType === VotingStrategyType.LINEAR_ERC20) {
+      if (this.parentStrategyType === VotingStrategyType.LINEAR_ERC20 || this.parentStrategyType === VotingStrategyType.LINEAR_ERC20_HATS_WHITELISTING) {
         freezeVotingByteCodeLinear = generateContractByteCodeLinear(
           this.freezeVotingErc20MasterCopy,
         );
-      } else if (this.parentStrategyType === VotingStrategyType.LINEAR_ERC721) {
+      } else if (this.parentStrategyType === VotingStrategyType.LINEAR_ERC721 || this.parentStrategyType === VotingStrategyType.LINEAR_ERC721_HATS_WHITELISTING) {
```
Comment (DarksightKellar) [inline src/models/FreezeGuardTxBuilder.ts:220]: Might as well also extract these booleans to `isXYZType` consts, no?

**Comment by adamgall** [comment]
@mudrila can we get this branch merged into `develop`, with whatever necessary feature-flag-gating needs to be added to keep the new ui out of the eyes of users?

File: src/components/Roles/forms/RoleFormInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+            {({
+              field,
+              form: { setFieldTouched },
+              meta,
+            }: {
+              field: FieldInputProps<string>;
+              form: FormikProps<RoleFormValues>;
+              meta: FieldMetaProps<string>;
+            }) => (
+              <TextareaComponent
```
[the remark this answers is not in the document]
Comment (mudrila) [reply src/components/Roles/forms/RoleFormInfo.tsx:80]: Nope - we're using `TextareaComponent` across the app, which is our own implementation using `LabelComponent` and `Input`

**Review by adamgall — APPROVED** [review]
LGTM

File: src/hooks/utils/useCreateRoles.ts
Comment (DarksightKellar) [inline src/hooks/utils/useCreateRoles.ts:1]: Not for this PR, just noting that I'd love to see this file trimmed down sometime soon -- move feature-specific code to their own feature (whitelisting, termed roles) files, and keep this file as lean as possible and focused on _role creation_ itself

**Review by DarksightKellar — APPROVED** [review]
lgtm

**Pull request merged**